bot-components/String_utils.ml

```diff
@@ -0,0 +1,146 @@
+(******************************************************************************)
+
+let strip_quoted_bot_name ~github_bot_name body =
+  (* If someone says "`@coqbot minimize foo`", (with backticks), we
+     don't want to treat that as them tagging coqbot, so we adjust
+     the tagging to "@`coqbot minimize foo`" so that the matching
+     below doesn't pick up the name *)
+  Str.global_replace
+    (Str.regexp
+       (Printf.sprintf "\\(`\\|<code>\\)@%s:? " (Str.quote github_bot_name)) )
+    (Printf.sprintf "@\\1%s " (Str.quote github_bot_name))
+    body
+
+let%expect_test "strip_quoted_bot_name" =
+  Stdio.printf "%s\n"
+    (strip_quoted_bot_name ~github_bot_name:"coqbot"
+       {|>this didn't produce a pipeline for some reason\r\n\r\nI think that this is normal. @herbelin was maybe expecting that adding the `request: full CI` label would trigger a new run immediately, but the semantics is that this label will produce such a full CI run at the next update (next push) of this PR. Cf. the [documentation](https://github.com/coq/coq/blob/master/CONTRIBUTING.md#understanding-automatic-feedback):\r\n\r\n>you can request a full run of the CI by putting the `request: full CI` label before pushing to your PR branch, or by commenting `@coqbot: run full CI` after having pushed. |} ) ;
+  [%expect
+    {| >this didn't produce a pipeline for some reason\r\n\r\nI think that this is normal. @herbelin was maybe expecting that adding the `request: full CI` label would trigger a new run immediately, but the semantics is that this label will produce such a full CI run at the next update (next push) of this PR. Cf. the [documentation](https://github.com/coq/coq/blob/master/CONTRIBUTING.md#understanding-automatic-feedback):\r\n\r\n>you can request a full run of the CI by putting the `request: full CI` label before pushing to your PR branch, or by commenting @`coqbot run full CI` after having pushed. |}]
+
+let clean_gitlab_trace trace =
+  trace
+  |> Str.global_replace (Str.regexp "\027\\[[0-9;]*m") ""
+  |> Str.global_replace (Str.regexp "\027\\[0K") ""
+  |> Str.global_replace (Str.regexp "section_start:[0-9]*:[a-z_]*\r") ""
+  |> Str.global_replace (Str.regexp "section_end:[0-9]*:[a-z_]*\r") ""
+  |> String.split_lines
+
+let shorten_ci_check_name target =
+  target
+  |> Str.global_replace (Str.regexp "GitLab CI job") ""
+  |> Str.global_replace (Str.regexp "(pull request)") ""
+  |> Str.global_replace (Str.regexp "(branch)") ""
+  |> Stdlib.String.trim
+
+(******************************************************************************)
+(* Hashtable Formatting                                                       *)
+(******************************************************************************)
+
+let string_of_mapping mapping =
+  Hashtbl.fold ~init:""
+    ~f:(fun ~key ~data acc -> acc ^ Printf.sprintf "(%s, %s)\n" key data)
+    mapping
+
+let string_of_conclusion conclusion =
```

Comment:
Despite the name, I'm unsure it is actually a string util, and also unsure about moving it away from where it is used and exposing it in the API.